internal/provider/indices_resource.go

```diff
@@ -519,6 +519,48 @@ func (r *indicesResource) findAndCreateState(indices []go_marqo.IndexDetail, ind
 	return nil, false
 }
 
+func normalizeIndexSettings(state *IndexResourceModel, newState *IndexResourceModel) {
```

Comment:
This feels very brittle to changes in schema. In the worst case, adding new fields could potentially cause existing indexes to be deleted if we don't update this function.

Feels like it should have a more systematic solution, i.e. walk the `newState` fields and descendant fields, and propagate nullish values.

Reply:
I thought so as well. So I'm not super happy with it yet. 

problems:
1. user does not define these fields so they are null
2. api returns default values
3. mismatch between null and the default values

Manual Solutions I've tried
1. set incoming fields to null if they are null locally (not ideal, want to show full state to user)
2. set undefined fields with the default value before create (not good, default value drift from cloud is a possibility)

Both don't seem appealing.

I'm looking for a way to indicate to terraform that some fields will be undefined in the user's statefile but should be expected from the api. Looking through the documentation I've found [this](https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification) which seems to fit the function